test/test_config.py

```diff
@@ -0,0 +1,181 @@
+                f"Executable for {model['name']} submodel should be " +
+                f"an absolute path: {config['exe']}"
+            )
+
+    def test_exe_paths_in_manifest(self, config, exe_manifest_fullpaths):
+        if 'exe' in config:
+            assert config['exe'] in exe_manifest_fullpaths, (
+                f"Model executable path should be in Manifest file " +
+                f"(e.g. manifests/exe.yaml): {config['exe']}"
+            )
+
+    def test_sub_model_exe_paths_in_manifest(self, config,
+                                             exe_manifest_fullpaths):
+        for model in config.get('submodels', []):
+            if 'exe' in model:
+                assert model['exe'] in exe_manifest_fullpaths, (
+                    f"Submodel {model['name']} executable path should be in " +
+                    f"Manifest file (e.g. manifests/exe.yaml): {config['exe']}"
+                )
+
+    def test_restart_freq_is_date_based(self, config):
+        assert "restart_freq" in config, "Restart frequency should be defined"
+        frequency = config["restart_freq"]
+        # String of an integer followed by a YS/MS/W/D/H/T/S unit,
+        # e.g. 1YS for 1 year-start
+        pattern = r'^\d+(YS|MS|W|D|H|T|S)$'
+        assert isinstance(frequency, str) and re.match(pattern, frequency), (
+           "Restart frequency should be date-based: " +
+           f"'restart_freq: {frequency}'"
+        )
+
+    def test_sync_is_not_enabled(self, config):
+        if 'sync' in config and 'enable' in config['sync']:
+            assert not config['sync']['enable'], (
+                "Sync to remote archive should not be enabled"
+            )
+
+    def test_sync_path_is_not_set(self, config):
+        if 'sync' in config:
+            assert not ('path' in config['sync']
+                        and config['sync']['path'] is not None), (
+                "Sync path to remote archive should not be set"
+            )
+
+    def test_manifest_reproduce_exe_is_on(self, config):
+        manifest_reproduce = config.get('manifest', {}).get('reproduce', {})
+        assert 'exe' in manifest_reproduce and manifest_reproduce['exe'], (
+            "Executable reproducibility should be enforced, e.g set:\n" +
+            "manifest:\n    reproduce:\n        exe: True"
+            )
+
+    def test_no_scripts_in_top_level_directory(self, control_path):
+        exts = {".py", ".sh"}
+        scripts = [p for p in control_path.iterdir() if p.suffix in exts]
+        assert scripts == [], (
+            "Scripts in top-level directory should be moved to a " +
+            "'tools' sub-directory"
+        )
+
+    def test_validate_metadata(self, metadata):
+        # Get schema from Github
+        schema_version = metadata.get("schema_version", DEFAULT_SCHEMA_VERSION)
+        schema_path = f"{BASE_SCHEMA_PATH}/{schema_version}.json"
+    
+        url = f"{BASE_SCHEMA_URL}/main/{schema_path}"
```

Comment:
Sorry to be a pain, but I thought we were trying to standardise on referencing schema by commit or tag rather than using `main`? In case the path changes in the future the URL is immutable and will remain valid.

Reply:
It is a different use-case. It's not embedded in code that might be deployed, and can be updated, so no stress if it is much more difficult, or not really applicable.

Reply:
I've decided it isn't really applicable, but happy to change it at a later date if we do decide it is worthwhile.

Reply:
Yeah I went with 'main' so if metadata has a later schema version specified, it'll be able to find that schema. If it can't find the schema, it'll default to use the default schema version `1-0-0` at a fixed commit